src/wallet/wallet2.cpp

```diff
@@ -6017,6 +6017,18 @@ std::map<uint32_t, uint64_t> wallet2::balance_per_subaddress(uint32_t index_majo
         found->second += utx.second.m_change;
     }
    }
+
+   for (const auto& utx: m_unconfirmed_payments)
+   {
+    if (utx.second.m_pd.m_subaddr_index.major == index_major)
+    {
+      auto found = amount_per_subaddr.find(utx.second.m_pd.m_subaddr_index.minor);
+      if (found == amount_per_subaddr.end())
+        amount_per_subaddr[utx.second.m_pd.m_subaddr_index.minor] = utx.second.m_pd.m_amount;
+      else
+        found->second += utx.second.m_pd.m_amount;
+    }
```

Comment:
These 5 lines can be just:
amount_per_subaddr[utx.second.m_pd.m_subaddr_index.minor] += utx.second.m_pd.m_amount;
since the new element is zero initialized.